trunner/long_test.py

```diff
@@ -0,0 +1,140 @@
+#
+# Phoenix-RTOS test runner
+#
+# Long Test Module - Test Runner support for the phoenix-rtos-ports test suites
+#
+# Copyright 2022 Phoenix Systems
+# Authors: Damian Loewnau
+#
+
+from .tools.color import Color
+
+
+LONG_TESTS = ['busybox', 'mbedtls']
+
+
+class LongTestHarnessFactory:
+    @staticmethod
+    def create(ltest_type):
+        if (ltest_type == 'busybox'):
+            return BusyboxTestHarness.harness
+        if (ltest_type == 'mbedtls'):
+            return MbedtlsTestHarness.harness
+        else:
+            raise ValueError(f"Unknown long test type: {ltest_type}")
+
+
+class LongTestResult:
+    """Class representing results of one of the long test suites."""
+
+    PASS = 'PASS'
+    FAIL = 'FAIL'
+    IGNORE = 'SKIPPED'
+
+    def __init__(self, name, status, msg=''):
+        self.name = name
+        self.status = status
+        self.msg = msg
+
+    def __str__(self):
+        if self.status == LongTestResult.PASS:
+            color = Color.OK
+        elif self.status == LongTestResult.FAIL:
+            color = Color.FAIL
+        elif self.status == LongTestResult.IGNORE:
+            color = Color.SKIP
+
+        status = Color.colorify(self.status, color)
+
+        res = f"{status}: {self.name}"
+        if self.status == LongTestResult.FAIL:
+            res += '\n\t\tOutput of the failed test case:\n\t\t---\n'
+            res += f'{self.msg}'
+            res += '\t\t---'
+        return res
+
+
+class MbedtlsTestHarness:
+    """Class providing harness for parsing output of Mbedtls tests"""
+
+    # common result line: 'Hash: RIPEMD160 ................................................... PASS'
+    RESULT = r"(?P<test_name>.{67})\s(?P<status>(PASS|----|FAILED))\r+\n"
+    # after failed test case there is some additional output
+    MSG_LINE = r"  ([^\r\n]+?)\r+\n"
+    # common final line: 'FAILED (16 / 21 tests (6 skipped))'
+    FINAL = r"(?P<status>FAILED|PASSED)\s\((?P<nr>\d+)\s/\s(?P<total_nr>\d+)\stests\s\(\d+\sskipped\)\)"
```

Comment:
Please add test cases to `trunner/test` that covers `MbedtlsTestHarness.harness` method. Now the long test are not running up on PRs so I can't even check how the output of Mbedtls tests looks like without setting up the zynq target with ports. This is also a good way to document the cases of the Mbedtls tests output (e.g. passed tests, failed tests with msg, failed tests with multiline message, skipped tests)